tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -42,293 +49,789 @@
    },
    "outputs": [],
    "source": [
-    "# Plot all the bin trips on the map, later we should use different color for different bins, but not sure whether it works.\n",
-    "m = folium.Map(location=[trips[0].data.start_loc[\"coordinates\"][1], trips[0].data.start_loc[\"coordinates\"][0]],zoom_start=12, max_zoom= 30, control_scale=True)\n",
-    "t_index = 0\n",
-    "for bin in sim.bins:    \n",
-    "    for i in range (len(bin)):\n",
-    "        layer = folium.PolyLine([[trips[t_index].data.start_loc[\"coordinates\"][1],trips[t_index].data.start_loc[\"coordinates\"][0]],\n",
-    "                                [trips[t_index].data.end_loc[\"coordinates\"][1],trips[t_index].data.end_loc[\"coordinates\"][0]]],weight=2, color='orange')\n",
-    "        t_index += 1\n",
-    "        layer.add_to(m)\n",
-    "        \n",
-    "m"
+    "#Plot the clusters above the cutoff only with different colors,to see if the algorithm wrongly puts different trips into the same cluster\n",
+    "cluster_cutoff = folium.Map(location=[feat.points[0][1], feat.points[0][0]],zoom_start=12, max_zoom= 30, control_scale=True)\n",
+    "\n",
+    "print(feat.labels)\n",
+    "if feat.labels:             \n",
+    "    for i in range(len(feat.points)):\n",
+    "        start_lat = feat.points[i][1]\n",
+    "        start_lon = feat.points[i][0]\n",
+    "        end_lat = feat.points[i][3]\n",
+    "        end_lon = feat.points[i][2]\n",
+    "        layer = folium.PolyLine([[start_lat,start_lon],\n",
+    "                                 [end_lat,end_lon]],weight=2, color=cmp_clusters_cutoff(feat.labels[i]))\n",
+    "        layer.add_to(cluster_cutoff)\n",
+    "cluster_cutoff.add_child(cmp_clusters_cutoff)    \n",
+    "cluster_cutoff"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "better-continent",
+   "id": "cloudy-language",
    "metadata": {},
    "source": [
-    "### Trips from bins above the cutoff only"
+    "## Visualizing the difference between bins and clusters above cutoff "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "greenhouse-roberts",
+   "metadata": {},
+   "source": [
+    "Note: this part can be run alone, after importing packages"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "round-death",
+   "id": "forced-gasoline",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logger = logging.getLogger()\n",
+    "logger.setLevel(logging.DEBUG)\n",
+    "participant_uuid_obj = list(edb.get_profile_db().find({\"install_group\": \"participant\"}, {\"user_id\": 1, \"_id\": 0}))\n",
+    "all_users = [u[\"user_id\"] for u in participant_uuid_obj]\n",
+    "radius = 300"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "second-instrumentation",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We use user11 as an example\n",
+    "#-above cutoff only\n",
+    "#DEBUG:root:number of bins after filtering: 14\n",
```

Comment:
can we remove these log statements? Why are they useful?
Isn't this it the same as including outputs in the notebook?